pymepix/processing/logic/processing_step.py

```diff
@@ -0,0 +1,18 @@
+from abc import abstractmethod, ABC
+
+from pymepix.core.log import Logger
+
+class ProcessingStep(Logger, ABC):
+
```

Comment:
Added  the docstring and tried to explain the connection among the classes briefly. Not sure whether a diagram would be helpful because I still do not fully understand all t he things happening in the (already) existing pipeline implementation. Therefore the picture would be rather incomplete. For The additions I implemented the brief explanation should satisfy the need for documentation.

Reply:
TODO: @troehling, @BenMoon create a diagram to explain this completely #25